server/controllers/card.js

```diff
@@ -0,0 +1,51 @@
+const Card = require("../models/Card");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /users
+// @desc Search for users
+// @access Private
+exports.editCard = asyncHandler(async (req, res, next) => {
+
+  const { _id } = req.params;
+
+  const {
+    name,
+    description,
+    colorCode,
+    deadline,
+    tags,
+    attachments
+  } = req.body;
+
+  try {
+    await Card.findOneAndUpdate({_id}, {
+      name,
+      description,
+      colorCode,
+      deadline,
+      tags,
+      attachments
+    });
+  } catch (err) {
+    res.status(404)
+    throw new Error("No card found with given id")
+  }
```

Comment:
So, async-express-handler actually handles catching any errors

Reply:
Also, if no card is found then it will return null and not throw an error.

Reply:
Yes, I saw that when I refactor the card delete. but I forgot to fix on cardEdit. thanks @moffatethan  I already fixed this issue.